igor-gateway/src/config.rs

```diff
@@ -1,15 +1,58 @@
 use std::net::SocketAddr;
-use uclicious::Uclicious;
+use uclicious::{Uclicious};
+use std::path::{Path, PathBuf};
+use std::error::Error;
+use uclicious::{DEFAULT_DUPLICATE_STRATEGY, Priority};
 
-#[derive(Debug, Uclicious)]
-#[ucl(include(chunk = ""))]
+#[derive(Debug, Uclicious, Clone)]
 pub struct Config {
-    #[ucl(default = "String::from(\"sqlite::memory:\")")]
-    pub database_url: String,
-    #[ucl(default = "listen_default()")]
+    pub database_path: String,
     pub listen: SocketAddr,
+    pub mqtt: Mqtt,
 }
 
-fn listen_default() -> SocketAddr {
-    "127.0.0.1:1337".parse().unwrap()
+#[derive(Debug, Uclicious, Clone)]
+#[ucl(skip_builder)]
+pub struct Mqtt {
+    pub host: SocketAddr,
+    pub authentication: Option<MqttAuthentication>,
 }
+
+#[derive(Debug, Uclicious, Clone)]
+#[ucl(skip_builder)]
+pub struct MqttAuthentication {
+    pub username: String,
+    pub password: String,
+}
+
+impl Config {
+    pub fn ensure_paths(&self) -> Result<(), Box<dyn Error>> {
+        {
+            if let Some(path) = Path::new(&self.database_path).parent() {
```

Comment:
wait where is path defined?

Reply:
Right on this line.

https://doc.rust-lang.org/stable/rust-by-example/flow_control/if_let.html